cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```diff
@@ -66,12 +71,17 @@ public void onReceived(Iterator<FetchedMessage> messages) {
 
     int count = 0;
 
+    long oldestProcessed = 0;
     while (messages.hasNext()) {
       FetchedMessage message = messages.next();
       try {
         GenericRecord genericRecord = serializer.toGenericRecord(message.getPayload());
         ILoggingEvent event = serializer.fromGenericRecord(genericRecord);
 
+        if (event.getTimeStamp() < oldestProcessed) {
```

Comment:
with the current code we show what is the lag based on the fetched log messages from kafka. I believe it is fine.